app/routes.py

```diff
@@ -1 +1,275 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint,jsonify, abort, make_response, request
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime 
+import os
+import requests
+
+task_bp = Blueprint("task_bp",__name__, url_prefix="/tasks")
+
+def get_one_task_or_abort(task_id):
+    try:
+        task_id = int(task_id)
+    except ValueError:
+        response_str = f"Invalid task_id: `{task_id}`.Id must be an integer."
+        abort(make_response(jsonify({'message':response_str}), 400))
+
+    matching_task = Task.query.get(task_id)
+
+    if not matching_task: 
+        response_str = f"Task with id {task_id} was not found in the database."
+        abort(make_response(jsonify({'message':response_str}), 404))
+    return matching_task
+
+
+@task_bp.route("", methods=["POST"])
+def add_task():
+    request_body = request.get_json()
+
+    if "title" not in request_body or \
+        "description" not in request_body:
+            return jsonify({"details": "Invalid data"}), 400
+
+    new_task = Task(
+        title = request_body["title"],
+        description = request_body["description"])
+
+    db.session.add(new_task)
+
+    db.session.commit()
+
+    return {
+        "task":{
+            "id" : new_task.id,
+            "title" : new_task.title,
+            "description" : new_task.description,
+            "is_complete" : new_task.is_complete
+        }
+    }, 201
+
+@task_bp.route("", methods=["GET"])
+def get_all_tasks():
+    title_param = request.args.get("title")
+
+    sort_param = request.args.get("sort")
+
+    if not title_param:
+        tasks = Task.query.all()
+    
+    if sort_param  == "desc":
+        tasks = Task.query.order_by(Task.title.desc()).all()
+
+    if sort_param == "asc":
+        tasks = Task.query.order_by(Task.title.asc()).all()
```

Comment:
This might cause `Task.query` to execute twice making this potentially twice as slow as it needs to be.